### PR TITLE
Viewer service refactoring

### PR DESCRIPTION
**Pull request by alabiaga**
extract canRenderTemplates and navigateToAmpUrl from viewer service.

fixes 2/3 tasks from #17590

### CODE REVIEW

**Comment by alabiaga** [comment]
@lannka moving out isTrustedReferrer wasn't as trivial as the other changes. I'd also like to read into it to more to understand what it does. Will follow up tonight or this week with that particular refactoring.

File: src/ssr-template-helper.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ export class SsrTemplateHelper {
    * @return {boolean}
    */
   isSupported() {
-    return this.viewer_.canRenderTemplates();
+    return this.viewer_.hasCapability(Capability.VIEWER_RENDER_TEMPLATE);
```
Comment (choumx) [inline src/ssr-template-helper.js]: The `Capability` enum should either have all of the capabilities listed in CAPABILITIES.md or be removed outright.

Comment (alabiaga) [reply src/ssr-template-helper.js]: removed..as much as I'd love to find all the the viewer capability types and create enums for them and replace them throughout the codebase.

File: test/functional/test-viewer.js [hunk 2]
```diff
@@ -1669,28 +1669,4 @@ describe('Viewer', () => {
...
-    });
-
-    it('should return false if a2a capability is not supported', () => {
-      const viewer = new Viewer(ampdoc);
-      const send = sandbox.stub(viewer, 'sendMessage');
-      const result = viewer.navigateToAmpUrl(ampUrl, 'abc123');
-      expect(send).to.have.not.been.called;
-      expect(result).to.be.false;
-    });
-  });
```
[the remark this answers is not in the document]
Comment (alabiaga) [reply test/functional/test-viewer.js]: done. there were tests in test-navigation already but they the response from the navigateTo method call was stubbed entirely. fixed the existing tests.

File: build-system/tasks/bundle-size.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ const log = require('fancy-log');
 const {getStdout} = require('../exec');
 
 const runtimeFile = './dist/v0.js';
-const maxSize = '80.85KB';
+const maxSize = '80.9KB';
```
Comment (choumx) [inline build-system/tasks/bundle-size.js]: Why did this PR have a bundle size impact?

Comment (alabiaga) [reply build-system/tasks/bundle-size.js]: hmm good question. Perhaps the capability enum removal. Introducing the enum shorthanded the usage of the 'viewerRenderTemplate' in the codebase.

Comment (alabiaga) [reply build-system/tasks/bundle-size.js]: though i can't imagine it accounting for 5kb of data.

**Pull request merged**